forge16/src/main/java/com/envyful/poke/tracker/forge/tracker/PokeTrackerFactory.java

```diff
@@ -45,7 +45,13 @@ public static void addTrackedEntities(PixelmonEntity pixelmon) {
 
             List<EntityData> entities = TRACKED_ENTITIES.computeIfAbsent(entry.getValue().getName().toLowerCase(), ___ -> Lists.newArrayList());
 
-            entities.add(0, EntityData.of(pixelmon));
+
+            EntityData e = EntityData.of(pixelmon);
+
+            // Check if an Entity with the same UUID is already being tracked using lambda
+            if (!entities.stream().anyMatch(entityData -> entityData.getEntityUUID().equals(e.getEntityUUID()))) {
```

Comment:
Please don't use streams to check like this. Can you refactor this into it's own method and use a for loop.

Also, can you use Objects#equals for checking the UUIDs

Reply:
Thanks for your reply! Is there a reason to not use streams? Like performance or something like that.

Reply:
Yep, pretty much. Streams are notorious for being bad for performance and generally harder to read (as your comment shows, just reading a for loop or a function name, is far easier